docs/amber.md

```diff
@@ -0,0 +1,138 @@
+framebuffer is named `framebuffer`. The generated depth buffer is named
+`depth_buffer`. For each buffer seen we generate a name of `AutoBuf-<num>` where
+the number is the current number of buffers seen counting from 0.
+
+The VkScript parser is broken into three major chunks. The `Parser`,
+`SectionParser` and `CommandParser`. The `Parser` is the overall driver for the
+parser. The `SectionParser` breaks the input file into the overall chunks (each
+of the sections separated by the \[blocks]). The `CommandParser` parses the
+`[test]` section specifically. For other sections they're parsed directly in
+the `Parser` object.
+
+### Parsed Representation
+The `Script` object is the owner for all of our data. Other objects hold
+pointers but the script holds the `unique_ptr` for parsed objects.
+
+```
+                                        +--------+                 +--------------+
+                                        | Script |---------------->| Requirements |
+                                        +--------+                 +--------------+
+                                             |
+             +------------------+------------+--------+----------------------+
+             |                  |                     |                      |
+             v                  v                     v                      v
+      +---------+          +---------+        +---------+            +---------+
+      |  +--------+        |  +---------+     |  +----------+        |  +---------+
+      +--|  +--------+     +--|  +--------+   +--|  +----------+     +--|  +---------+
+         +--| Shader |        +--| Buffer |      +--| Pipeline |        +--| Command |
+            +--------+           +--------+         +----------+           +---------+
+               ^                  ^  ^                 |  |  ^               |    |
+               |                  |  |                 |  |  +---------------+    |
+Entry point    |                  |  +-----------------+  |                       |
+Optimizations  |                  |  Descriptor Set       |                       |
+Type           |                  |  Binding              |                       |
+Compiled Shader|                  |  Attachment Location  |                       |
+               |                  |                       |                       |
+               +------------------------------------------+                       |
+                                  |                                               |
+                                  +-----------------------------------------------+
+                                                        BufferCommand
+```
+
+A `Script` is made up for shaders, pipelines, buffers and commands. Pipelines
+contain shaders and buffers. The buffers hold both image and storage buffers.
+It is assumed that the `Buffer` object has the current value for that buffer at
```

Comment:
"Script execution assumes that after executing a command, each `Buffer` object has a reference to the latest data for that buffer or image mapped into host (CPU) memory."

Reply:
done. Although, it's not CPU memory, the buffer itself holds a chunk of memory the values have to be copied into.